src/python/espressomd/interactions.pyx

```diff
@@ -1700,6 +1700,15 @@ cdef class BondedInteraction:
     def __reduce__(self):
         return (self.__class__, (self._bond_id,))
 
+    def __getstate__(self):
+        # for most bonds, we only need to pickle arguments passed to the
+        # constructor, yet some bond classes need to pass additional data
+        return (self.params,)
+
+    def __setstate__(self, params):
+        # parameters are already set in the core by the contructor
```

Comment:
that comment is wrong.... the pickle interactions is in the ```__reduce__``` function, right? https://docs.python.org/3/library/pickle.html#object.__reduce__

Reply:
didn't realize `__reduce__` was used in pickle interactions, I'll revert these changes and use `__reduce__` directly

Reply:
changes reverted, counter is now checkpointed by `__reduce__` in 9c454a62aea